pengine/graph.c

```diff
@@ -1079,6 +1098,10 @@ action2xml(action_t * action, gboolean as_input, pe_working_set_t *data_set)
         add_downed_nodes(action_xml, action, data_set);
     }
 
+    if (safe_str_eq(action->task, CRM_OP_MAINTENANCE_NODES)) {
+        add_maintenance_nodes(action_xml, action, data_set);
+    }
+
```

Comment:
On the other hand of course when trying to optimize the most likely cases xml-crafting
is not done at all as long as there are no nodes in maintenance-mode.
Apart from the architectural uglyness crafting an xml-structure that holds just the
nodes with the maintenance-bit set inside some other node-loop would bring down
the average load but raise the worst-case-load probably.